scipy-1.0/paper.tex

```diff
@@ -185,6 +186,114 @@ \section*{Project organisation and community}
+and would benefit from the addition of the Bluestein algorithm. There's also
+too much community code duplication with \texttt{numpy.fft}, which
+requires clarification of best choices and long-term plans.
+
+\texttt{scipy.integrate} has a new ODE solver interface, and we should aim
+to gracefully deprecate the older API. Support for integrating complex-valued
+functions and integrating multiple intervals could be beneficial.
+
+\texttt{scipy.interpolate} could benefit from a set of new features including
+spline fitting routines with better user control; integration, differentiation,
+and arithmetic routines for splines; transparent tensor-product splines,
+Non-Uniform Rational B-Splines support; and mesh refinement and coarsening of
+B-splines and corresponding tensor products.
+
+\texttt{scipy.io} would benefit from PCM float support, but we otherwise
+direct users with more sophisticated needs to domain-specific audio
+libraries.
+
+For \texttt{scipy.linalg}, we need to remove functions that are duplicate with 
+\texttt{numpy.linalg} unless performance is superior in SciPy,
+\texttt{get\_lapack\_funcs} should always use flapack, and
+we should continue to wrap more LAPACK functions. Ideally, we'd also
+add type-generic wrappers in the Cython BLAS and LAPACK interfaces, and
+transform many of the linear algebra routines into gufuncs.
+
+We're planning to remove \texttt{scipy.misc} as a public module, and
+rename \texttt{scipy.odr} to \texttt{scipy.regression} or
+\texttt{scipy.fitting}, and relocate \texttt{optimize.curve\_fit}. This 
+module will then provide a home for other fitting functionality.
+\texttt{scipy.optimize} would benefit from a few more good global optimizers 
+as well as large-scale optimizers.
+
+\texttt{scipy.ndimage} should be completely migrated to a data point
+model (values on a grid), instead of a pixel model (elements with centers),
+both of which have been used in the past, because the data point model is now
+proving more effective in practice. We also need to standardize the
+\texttt{scipy.ndimage.morphology} API.
+
+\texttt{scipy.signal} needs its overlap with \texttt{scipy.ndimage}
+to be reduced; B-splines functionality should be moved to \texttt{interpolate},
+\texttt{firwin2} can be merged into \texttt{firwin}; Second Order Sections need
+to be updated to match the capabilities in other routines, and wavelet-related
+routines are somewhat out of place.
+
+While \texttt{scipy.sparse} is mature, there are many data
+formats and the community continues to move forward here so we will aim
+to take advantage of their improvements while preserving backwards
+compatibility to the extent that it is possible. \texttt{sparse.linalg}
+may benefit from wrappers for PROPACK for faster sparse SVD computation.
+
+For \texttt{scipy.spatial}, KDTree will be removed, and cKDTree will be 
+renamed to KDTree in a backwards-compatible way. \texttt{distance\_wrap.c} 
```

Comment:
I wouldn't mention the `KDTree` removal, users won't notice since it becomes an alias of `cKDTree`. There's nothing in `spatial` important enough to mention AFAIK.

Reply:
Sorry changed my mind: for `spatial` we can mention `spatial.transforms` with support for rotation matrices.